### PR TITLE
switch from ifc-core to reflifc

### PR DESCRIPTION
**Pull request by AndreyG**
I have created a little bit more high-level interface for work with IFC -- `ReflIFC`, wrapper around ifc-core. IMO it makes sense to switch from direct using of `ifc-core` to `reflifc`.

### CODE REVIEW

**Comment by FireFlyForLife** [comment]
Wow thanks! This wrapper API looks pretty clean in client code :).

I will be looking through the review in detail today. My first question is, what does `reflifc` stand for?

**Comment by AndreyG** [comment]
> My first question is, what does `reflifc` stand for?

It stands for "Reflection of IFC", actually name "reflifc" inspired by [this cool story](https://www.youtube.com/watch?v=ARxj3dfF_h0&t=2372s) from Andrew Sutton: `templarg` as ugliest identifier they made up for keyword.

File: tools/neat_code_gen/src/CodeGenerator.cpp [hunk 1]
```diff
@@ -173,107 +183,63 @@ CodeGenerator::TypeMembers CodeGenerator::render_members(std::string_view type_n
...
-			return rendered;
-		}
-		break;
-	default:
-		// TODO: Throw exception
-		assert(false && "Unexpected base class type sort");
-		return "";
+	for (auto type : bases)
+	{
+		rendered += render_base(type);
```
Comment (FireFlyForLife) [inline tools/neat_code_gen/src/CodeGenerator.cpp:206]: Damn this looks clean!

File: tools/neat_code_gen/src/CodeGenerator.cpp [hunk 2]
```diff
@@ -672,94 +607,65 @@ bool CodeGenerator::reflects_private_members(ifc::DeclIndex index)
...
 		return true;
 	case ifc::TypeSort::Designated:
-		{
-			const auto& designated_type = file.designated_types()[type_index];
-			return is_type_exported(designated_type.decl);
-		}
-		break;
-	case ifc::TypeSort::Method:
-		{
-			const auto& method = file.method_types()[type_index];
```
Comment (FireFlyForLife) [inline tools/neat_code_gen/src/CodeGenerator.cpp:689]: You removed this switch case here, but thinking about it, it should always be exported if I add forward declarations to NeatReflection.

**Review by FireFlyForLife — APPROVED** [review]

**Comment by FireFlyForLife** [comment]
Again, this looks good :) thanks for taking the time to implement this!

**Pull request merged**